elasticapm/contrib/serverless/aws.py

```diff
@@ -121,10 +121,13 @@ def __enter__(self):
         transaction_type = "request"
         transaction_name = os.environ.get("AWS_LAMBDA_FUNCTION_NAME", self.name)
 
-        if "httpMethod" in self.event:  # API Gateway
+        self.httpmethod = self.event.get("requestContext", {}).get(
```

Comment:
I added a utility function to get deeply nested dict keys with one of my recent PRs:

https://github.com/elastic/apm-agent-python/blob/a06f74f6da75554bed0ea609f3f327721108674e/elasticapm/utils/__init__.py#L200-L207

I should also add a docstring that explains the usage, but it should be something like

    self.httpmethod = nested_key(self.event, "requestContext", "httpMethod") or nested_key(self.event, "requestContext", "http", "method")

Reply:
Oh right! Thank you I had forgotten that.